### ISSUE TITLE
Confusing behavior when using inline comment to disable Lint/EmptyBlock

### ISSUE THREAD

**Issue by mvz** [rubocop-hq/rubocop#8971]
I have the following test in one of my projects:

```ruby
  def test_empty_combinator
    assert_equal nil, combinator {}.call
  end
```

This triggers Lint/EmptyBlock. If I try to disable this using the regular method of disabling cops in code, the result is surprising. I use a comment like so:

```ruby
  def test_empty_combinator
    assert_equal nil, combinator {}.call # rubocop:disable Lint/EmptyBlock
  end
```

## Expected behavior

The offense goes away and there is are no more offenses on this line

## Actual behavior

RuboCop complains that disabling the cop is not necessary:

```
For /home/matijs/Projects/iba: configuration from /home/matijs/Projects/iba/.rubocop.yml
configuration from /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rubocop-performance-1.8.1/config/default.yml
configuration from /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rubocop-performance-1.8.1/config/default.yml
Default configuration from /home/matijs/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rubocop-1.1.0/config/default.yml
Inheriting configuration from /home/matijs/Projects/iba/.rubocop_todo.yml
Inspecting 1 file
Scanning /home/matijs/Projects/iba/test/call_test.rb
For /home/matijs/Projects/iba/test: configuration from /home/matijs/Projects/iba/test/.rubocop.yml
Inheriting configuration from /home/matijs/Projects/iba/.rubocop.yml
Inheriting configuration from /home/matijs/Projects/iba/.rubocop_todo.yml
AllCops/Exclude configuration from /home/matijs/Projects/iba/.rubocop.yml
Inheriting configuration from /home/matijs/Projects/iba/.rubocop_todo.yml
W

Offenses:

test/call_test.rb:8:42: W: Lint/RedundantCopDisableDirective: Unnecessary disabling of Lint/EmptyBlock.
    assert_equal nil, combinator {}.call # rubocop:disable Lint/EmptyBlock
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```
## Analysis

I think what happens is that Lint/EmptyBlock does not register an offense if there is some comment associated with the empty block. So the mere presence of the comment already causes the offense to go away. The RedundantCopDisableDirective cop then kicks in because no EmptyBlock offense is found anymore.

## Steps to reproduce the problem

Create a file with the code above and run RuboCop 1.1.0 on it.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.1.0 (using Parser 2.7.2.0, rubocop-ast 1.1.0, running on ruby 2.7.1 x86_64-linux)
  - rubocop-performance 1.8.1
```
